src/backend/InvenTree/common/serializers.py

```diff
@@ -638,3 +638,52 @@ class IconPackageSerializer(serializers.Serializer):
     prefix = serializers.CharField()
     fonts = serializers.DictField(child=serializers.CharField())
     icons = serializers.DictField(child=IconSerializer())
+
+
+class SelectionEntrySerializer(InvenTreeModelSerializer):
+    """Serializer for a selection entry."""
+
+    class Meta:
+        """Meta options for SelectionEntrySerializer."""
+
+        model = common_models.SelectionListEntry
+        fields = '__all__'
+
+    def validate(self, attrs):
+        """Ensure that the selection list is not locked."""
+        ret = super().validate(attrs)
+        if self.instance and self.instance.list.locked:
+            raise serializers.ValidationError({'list': _('Selection list is locked')})
+        return ret
+
+
+class SelectionListSerializer(InvenTreeModelSerializer):
+    """Serializer for a selection list."""
+
+    class Meta:
+        """Meta options for SelectionListSerializer."""
+
+        model = common_models.SelectionList
+        fields = [
+            'pk',
+            'name',
+            'description',
+            'active',
+            'locked',
+            'source_plugin',
+            'source_string',
+            'default',
+            'created',
+            'last_updated',
+            'choices',
+        ]
+
+    default = SelectionEntrySerializer(read_only=True, many=False)
+    choices = SelectionEntrySerializer(source='entries', read_only=True, many=True)
```

Comment:
@SchrodingersGat any tips on how serializing the choices could be done here in a way that one can update and create entries through that serializer/endpoint? This is the only thing missing in this PR any I can not wrap my head around it.

Reply:
Do you mean performing a bulk-create operation via the child serializer? I am not sure how that could be done - or if it would be possible. I think it would be going against the grain of how similar API endpoints work currently. Should we not have a separate API endpoint for accessing and updating the individual choices?

Reply:
That was the idea to keep the API simpler; I will look into alternative routes